portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,17 +16,56 @@
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    private List<String> comments = new ArrayList<String>();
+
+    @Override 
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+        response.setContentType("application/json;");
+        response.getWriter().println((json));    
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String comment = getComment(request);
+
+        // Add comment to comments Array.
+        comments.add(comment);
+
+        // Sends JSON to the client.
+        String json = gson.toJson(comments);
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+
+        // Redirect back to the HTML page.
+        response.sendRedirect("/index.html");  
+        }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getComment(HttpServletRequest request) {
+    //   Get comment from form.
+    String comment = request.getParameter("comment");
+    if(comment == null){
+        return null;
+    }
+    return request.getParameter("comment");
```

Comment:
This can be the entire method body. If getParameter returns null then getComment will return null here as desired.

Reply:
I see so there is no need for error checking due to it already having our desired behaviour initially ( Returning null inherently) . The thing is that we should check the value of the usages of getComment ?

Reply:
Yeah exactly. Arguably this method can be removed / inlined now since it's only a single line. The call site that builds the new list of comments needs to handle the case where a comment can't be pulled out of the request.

Reply:
Gotcha done